core/src/helpers.rs

```diff
@@ -169,16 +172,290 @@ pub fn starknet_block_to_eth_block(block: MaybePendingStarknetBlock) -> RichBloc
+            ether_tx.hash = H256::from_slice(&declare_tx.transaction_hash.to_bytes_be());
+            ether_tx.nonce = field_element_to_u256(declare_tx.nonce);
+            ether_tx.from = starknet_address_to_ethereum_address(declare_tx.sender_address);
+            // Define gas_price data
+            ether_tx.gas_price = None;
+            // Extracting the signature
+            ether_tx.r = field_option_element_to_u256(declare_tx.signature.get(0));
+            ether_tx.s = field_option_element_to_u256(declare_tx.signature.get(1));
+            ether_tx.v = field_option_element_to_u256(declare_tx.signature.get(2));
+            // Extracting the to address
+            ether_tx.to = None;
+            // Extracting the value
+            ether_tx.value = U256::from(0);
+            // Extracting the gas
+            ether_tx.gas = U256::from(0);
+            // Extracting the chain_id
+            ether_tx.chain_id = Some(1263227476_u64.into());
+            // Extracting the standard_v
+            ether_tx.standard_v = U256::from(0);
+            // Extracting the public_key
+            ether_tx.public_key = None;
+        }
+        StarknetTransaction::Deploy(deploy_tx) => {
+            // Extract relevant fields from InvokeTransactionV0 and convert them to the corresponding fields in EtherTransaction
+            ether_tx.hash = H256::from_slice(&deploy_tx.transaction_hash.to_bytes_be());
+            // Define gas_price data
+            ether_tx.gas_price = None;
+
+            ether_tx.creates = None;
+            // Extracting the public_key
+            ether_tx.public_key = None;
+        }
+        StarknetTransaction::DeployAccount(deploy_account_tx) => {
+            ether_tx.hash = H256::from_slice(&deploy_account_tx.transaction_hash.to_bytes_be());
+            ether_tx.nonce = field_element_to_u256(deploy_account_tx.nonce);
+            // TODO: Get from estimate gas
+            ether_tx.gas_price = None;
+            // Extracting the signature
+            ether_tx.r = field_option_element_to_u256(deploy_account_tx.signature.get(0));
+            ether_tx.s = field_option_element_to_u256(deploy_account_tx.signature.get(1));
+            ether_tx.v = field_option_element_to_u256(deploy_account_tx.signature.get(2));
+            // Extracting the to address
+            ether_tx.to = None;
+            // Extracting the gas
+            ether_tx.gas = U256::from(0);
+            // Extracting the chain_id
+            ether_tx.chain_id = Some(1263227476_u64.into());
+            // Extracting the standard_v
+            ether_tx.standard_v = U256::from(0);
+            // Extracting the public_key
+            ether_tx.public_key = None;
+        }
+    }
+
+    ether_tx
+}
+
+fn field_option_element_to_u256(element: Option<&FieldElement>) -> U256 {
+    match element {
+        Some(x) => {
+            let inner: u64 = x.to_string().parse().unwrap();
```

Comment:
this might panic:) 
no `unwrap` and no `u64` intermediary format 

In rust, u64 is 64 bits, whereas felt is potentially 251 bits
this means we should never use u64 as intermediary format but only bytes

Reply:
Didnt know that! Changed to transforming to bytes and then to U256 :)

Reply:
Done